shared/chat/conversation/list-hoc.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import * as ChatConstants from '../../constants/chat'
+import type {Props} from './list'
+import type {Options} from './messages'
+import {withProps} from 'recompose'
+
+type OptionsFn = (message: ChatConstants.Message, prevMessage: ChatConstants.Message, isFirstMessage: boolean, isSelected: boolean, isScrolling: boolean, key: any, style: Object, onAction: () => void) => Options
+
+function propsToMessageOptionsFn (props: Props): OptionsFn {
+  return function (message, prevMessage, isFirstMessage, isSelected, isScrolling, key, style, onAction): Options {
+    const skipMsgHeader = (message.author != null && prevMessage && prevMessage.type === 'Text' && prevMessage.author === message.author)
+    const isFirstNewMessage = message.messageID != null && props.firstNewMessageID ? props.firstNewMessageID === message.messageID : false
+
+    const options = {
```

Comment:
:sheep: Perhaps pull the needed props elements out first so that most entries in here can be in `foo,` form without the colon?